data_structures/linked-lists.py

```diff
@@ -0,0 +1,53 @@
+class Node():
+  def __init__(self, value, next_node=None):
+    self.value = value
+    self.next_node = next_node
+
+
+class LinkedList():
+  def __init__(self, head_node_value):
+    self.head_node = Node(head_node_value)
+
+  def add_to_beginning(self, value):
+    new_head = Node(value)
+    new_head.next_node = self.head_node
+    self.head_node = new_head
+
+  def stringify_list(self):
+    string = ''
+    current_node = self.head_node
+    while current_node:
+      string += f'{str(current_node.value)}\n'
+      current_node = current_node.next_node
+    return string
```

Comment:
There's nothing inherently _wrong_ with this code, but there are a few things that you can do to take advantage of Python a little better.

First, on line 20 above, you're casting a node to a string. This is generally fine, but if there's a general way that you're going to want a node to be displayed as a string, you can define a `__str__()` method on the Node class that will be automatically called whenever that object is being cast to a string. It's good practice to have a good `__str__` on your classes since printing them is so common.

Second, since this is essentially a string representation of the LinkedList, I would make this whole thing the `__str__` method for this class, since that's ultimately what you're trying to communicate. 

Conventionally, if you did that you'd also want to move it above `add_to_beginning()` since dunder methods (python class methods that are wrapped in double underscores) should be the first methods declared.

Finally, Since you're declaring, destroying, and then redeclaring a new string in your while loop, your space complexity is a little intensive. This isn't _wrong_ per say, but if you're comfortable with presenting a list you could essentially clean this code up like follows:

```python
def __str__(self):
    output = []
    current_node = self.head_node
    while current_node:
        output.append(current_node.value)
        current_node = current_node.next_node
    return str(output)
```

Alternatively, if you want to get _really_ fancy you could create a generator method called `gen_whole_list` and use it to generate that output list like so:

```python
def gen_complete_list(self):
    current_node = self.head_node
    
    while current_node:
        yeild current_node
        current_node = current_node
```

while would allow you to simplify your `__str__` method to a list comprehension like so:

```python
def __str__(self):
    return str([n.value for n in self.gen_whole_list()])
```

This gives you minimal space complexity since you're only ever creating one list which will contain references to existing nodes and not creating and then destroying strings over and over again for the garbage collector to pick up, and it still maintains your O(n) time complexity.